src/traces/isosurface/attributes.js

```diff
@@ -0,0 +1,125 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var colorAttrs = require('../../components/colorscale/color_attributes');
+var colorscaleAttrs = require('../../components/colorscale/attributes');
+var colorbarAttrs = require('../../components/colorbar/attributes');
+var mesh3dAttrs = require('../mesh3d/attributes');
+var baseAttrs = require('../../plots/attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+var attrs = {
+    x: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the x coordinates of the isosurface'
+        ].join(' ')
+    },
+    y: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the y coordinates of the isosurface'
+        ].join(' ')
+    },
+    z: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the z coordinates of the isosurface'
+        ].join(' ')
+    },
+
+    u: {
+        valType: 'data_array',
+        editType: 'calc',
+        description: 'Sets the intensity values of the isosurface.'
+    },
+
+    imin: {
+        valType: 'number',
+        editType: 'calc',
+        description: 'Sets the minimum iso bound of the isosurface.'
+    },
+
+    imax: {
+        valType: 'number',
+        editType: 'calc',
+        description: 'Sets the maximum iso bound of the isosurface.'
+    },
```

Comment:
`imax`/`imin` is OK if we rename `u` to `intensity`... but if it turns to `v` or `value` these should be `vmax`/`vmin`. I presume that if you only provide one of these, the other one is treated as +/-Infinity for the purpose of constructing the isocaps?

Reply:
I remember us having the same concerns about _intensity_ in surface traces. We ended up naming that attribute [`surfacecolor`](https://github.com/plotly/plotly.js/blob/32e6984a46f5c295ca0814aa33d7f066c64ce5d5/src/traces/surface/attributes.js#L127-L132).

I think naming this field `u` or `v` would confusing as it is an array of scalars, not a component of a vector field. I like `value`, but maybe `volume` would better (that's how [MATLAB seems to call this field](https://www.mathworks.com/help/matlab/ref/isosurface.html))?

Reply:
About `imin` and `imax`, MATLAB calls it _isovalue_. Wouldn't `isomin` and `isomax` be better?